Vikta-API-testing-sample/src/test/java/com/test/api/user/UserTest.java

```diff
@@ -0,0 +1,195 @@
+                .when()
+                .log().all()
+                .get(loginPath)
+                .then()
+                .log().all()
+                .spec(defaultResponseSpec())
+                .extract()
+                .body().jsonPath().getList(".", UserDTO.class);
+
+        assertThat(user).isEmpty();
+    }
+
+    @Test
+    public void canSearchUserByTerm(){
+        final String searchPath = getTestEnvironment().getUserPath() + "/search";
+        final String termQueryParam = "term";
+        Random random = new Random();
+        List<UserDTO> users = getListOfUsers();
+
+        assertThat(users.size()).isGreaterThan(0);
+
+        int size = random.nextInt(users.size()-1) + 2;
+
+        String term =  users.stream()
+                .map(UserDTO::getLoginName)
+                .limit(size)
+                .map(user  -> user.replaceAll(" ", "~"))
+                .collect(Collectors.joining("|"));
+
+        List<UserDTO> resultList = given().spec(defaultRequestSpec())
+                .queryParam(termQueryParam, term)
+                .when()
+                .log().all()
+                .get(searchPath)
+                .then()
+                .log().all()
+                .spec(defaultResponseSpec())
+                .extract().body().jsonPath().getList(".", UserDTO.class);
+
+        assertThat(users)
+                .as("User list is empty, can't check if can find user")
+                .isNotEmpty();
+
+        assertThat(resultList)
+                .as("User search result list shouldn't be empty")
+                .isNotEmpty();
+
+        assertThat(resultList.size())
+                .as(String.format("Should find %s, but find only %s Users", size, resultList.size()))
+                .isEqualTo(size);
+    }
+
+    @Test
+    public void canGetUserBySurname(){
+        final String surnamePath = getTestEnvironment().getUserPath() + "/surname";
+        final String surnameQueryParam = "surname";
+        List<UserDTO> users = getListOfUsers();
+
+        String userSurname = users.get(new Random().nextInt(users.size())).getSurname();
+
+        List<UserDTO> resultList = given().spec(defaultRequestSpec())
+                .queryParam(surnameQueryParam, userSurname)
+                .when()
+                .log().all()
+                .get(surnamePath)
+                .then()
+                .log().all()
+                .spec(defaultResponseSpec())
+                .extract()
+                .body().jsonPath().getList(".", UserDTO.class);
+
+        assertThat(resultList.size())
+                .as("There should be no users with same login")
+                .isGreaterThan(0);
+
+        assertThat(resultList)
+                .as("Found user with wrong surname")
+                .allMatch(user -> user.getSurname().equals(userSurname));
+    }
+
+    private List<UserDTO> getListOfUsers(){
+        String pathToList = getTestEnvironment().getUserPath() + "/list";
```

Comment:
add usersPath property in test.properties - just as categoriesPath and refactor all use cases to get rid of using '+"\list"\' in tests